loader/loader.go

```diff
@@ -302,6 +302,25 @@ func Paths(path string, recurse bool) (paths []string, err error) {
 	return paths, err
 }
 
+// Dirs converts all filepaths to directories to use in the file watcher.
+// Some types of events will not register on the files themselves, so
+// watch the directory to prevent this problem
+func Dirs(paths []string) []string {
```

Comment:
This function isn't specific to the file watcher, right? If its just giving the set of parent directories for some list of paths we should update the comment to basically explain that.